spec/alephant/logger/json_spec.rb

```diff
@@ -1,137 +1,92 @@
 require 'date'
 require 'spec_helper'
 require 'alephant/logger/json'
+require 'alephant/logger/level'
+require_relative 'support/json_shared_examples'
 
 describe Alephant::Logger::JSON do
-  subject { described_class.new(log_path) }
-
   let(:fn)       { -> { 'foo' } }
   let(:log_path) { '/log/path.log' }
-  let(:log_file) { instance_double File }
+  let(:log_output_obj) { instance_double(File) }
+  let(:msg) { :write }
 
   before do
-    allow(File).to receive(:open) { log_file }
-    allow(log_file).to receive :sync=
+    allow(File).to receive(:open) { log_output_obj }
+    allow(log_output_obj).to receive :sync=
   end
 
-  shared_examples 'JSON logging' do
-    let(:log_hash) do
-      { 'foo' => 'bar', 'baz' => 'quux' }
-    end
-
-    it 'writes JSON dump of hash to log with corresponding level key' do
-      allow(Time).to receive(:now).and_return('foobar')
+  logging_levels = Alephant::Logger::LEVELS.map(&:to_s)
+  levels_size = logging_levels.size - 1
 
-      expect(log_file).to receive(:write) do |json_dump|
-        h = { 'timestamp' => 'foobar', 'uuid' => 'n/a', 'level' => level }
-        expect(JSON.parse(json_dump)).to eq h.merge log_hash
-      end
-
-      subject.send(level, log_hash)
-    end
-
-    it 'automatically includes a timestamp' do
-      expect(log_file).to receive(:write) do |json_dump|
-        t = JSON.parse(json_dump)['timestamp']
-        expect { DateTime.parse(t) }.to_not raise_error
-      end
-
-      subject.send(level, log_hash)
-    end
+  %w(debug info warn error).each_with_index do |level, i|
```

Comment:
Why don't we just reference the `LEVELS` constant we already have defined?

Reply:
@samfrench My intention was to keep the specs descriptive.